hibernate-core/src/test/java/org/hibernate/orm/test/batch/BatchUpdateAndVersionTest.java

```diff
@@ -136,7 +139,12 @@ public void testFailedUpdate(SessionFactoryScope scope) {
 			fail();
 		}
 		catch (OptimisticLockException ole) {
-			assertTrue( ole.getCause() instanceof StaleObjectStateException );
+			if (getDialect() instanceof MariaDBDialect ) {
```

Comment:
I guess this will fail with the old exception on older MariaDB versions even with the new driver? If so, this if condition needs an additional version check.

Reply:
it will works, but has been rewritten to be clear